File: 2025/November-30-2025/longestSub.py
# Longest Substring Without Repeating Characters
# Solved 
# Given a string s, find the length of the longest substring without duplicate characters.

# A substring is a contiguous sequence of characters within a string.

# Example 1:

# Input: s = "zxyzxyz"

# Output: 3


s = "zxyzxyz"

def longest_sub(s: str) -> int: 
  left = 0
  seen = set()
  max_length = 0

  for right in range(0, len(s)): 
    while s[right] in seen: 
      seen.remove(s[left])
      left+=1 
    seen.add(s[right])
    max_length = max(max_length, right - left + 1)
  return max_length


print(longest_sub(s))

File: 2025/October-25-2025/LongestSub.js

// Code
// Testcase
// Test Result
// Test Result
// Leet
// 3. Longest Substring Without Repeating Characters
// Solved
// Medium
// Topics
// premium lock icon
// Companies
// Hint
// Given a string s, find the length of the longest substring without duplicate characters.

 

// Example 1:

// Input: s = "abcabcbb"
// Output: 3

var s = "abcabcbb"

const longestSub = (s) => { 
  let maxLength = 0; 
  let set = new Set()
  let left = 0; 
  
  for(let right = 0; right < s.length; right++){
    while(set.has(s[right])){ 
      if(set.has(s[left])){ 
        set.delete(s[left])
      }
      left++
    }
    set.add(s[right])
    maxLength = Math.max(maxLength, right - left + 1)
  }
  return maxLength
}

console.log(longestSub(s))